pkg/component/gardener/access/access.go

```diff
@@ -30,25 +30,29 @@ import (
 const ManagedResourceName = "shoot-core-gardeneraccess"
 
 // New creates a new instance of the deployer for GardenerAccess.
+// `withShootAccess` parameter defines whether to create ClusterRoleBindings for shoot access (adminkubeconfig and viewerkubeconfig).
 func New(
 	client client.Client,
 	namespace string,
 	secretsManager secretsmanager.Interface,
 	values Values,
+	withShootAccess bool,
```

Comment:
1. Why this arg is not part of the Values struct? Is it kind of a value of the component?
2. Why don't we use a value like `IsGardenCluster`? Such a value is being used for [VPA](https://github.com/gardener/gardener/blob/80d9cdd6fa35c7014ab012ebfb53c02c7d3385ae/pkg/component/autoscaling/vpa/vpa.go#L92-L93), [blackbox-exporter](https://github.com/gardener/gardener/blob/2bd13829d9fe7f18f4b58653841fdce351b6c260/pkg/component/observability/monitoring/blackboxexporter/blackboxexporter.go#L68-L69), [plutono](https://github.com/gardener/gardener/blob/7808ab37c335b28a7606666c51a22885e3730bbc/pkg/component/observability/plutono/plutono.go#L115-L116), [vali](https://github.com/gardener/gardener/blob/1b425822b264e319274e40b486898454cbc6ff93/pkg/component/observability/logging/vali/vali.go#L117).